DependencyInjection/DoctrineExtension.php

```diff
@@ -186,7 +193,9 @@ protected function loadDbalConnection($name, array $connection, ContainerBuilder
     {
         $configuration = $container->setDefinition(sprintf('doctrine.dbal.%s_connection.configuration', $name), new ChildDefinition('doctrine.dbal.connection.configuration'));
         $logger        = null;
-        if ($connection['logging']) {
+
+        /** @psalm-suppress UndefinedClass */
+        if (! $this->isSfDebugMiddlewareAvailable() && $connection['logging']) {
```

Comment:
this condition looks wrong to me. the SQL logging might use the Middleware system as soon as the DBAL logging middleware is available, even if the profiler integration still uses the old logger interface. DBAL supports using both APIs at the same time.

Reply:
It's what is implemented, isn't it ? See https://github.com/doctrine/DoctrineBundle/blob/9cadf2c331b01a017482d8283b02b7edd6b82244/Tests/DependencyInjection/DoctrineExtensionTest.php#L1282, `isSfDebugMiddlewareAvailable` return true if Middleware interface is available but also the middleware for profiling provided by the Doctrine bridge.

Reply:
@l-vo can you address this concern?

Reply:
@stof ok, now I see what you mean :). Actually, here, if logging is enabled, a chain logger is used with `doctrine.dbal.logger` and `doctrine.dbal.logger.profiling`. If profiling middleware is available, logging middleware is also available and this chain has no reason to be created. Moreover, I must keep the condition on this line because below we have:
```
if ($logger) {
    $configuration->addMethodCall('setSQLLogger', [$logger]);
}
```
Is it ok for you ?

Reply:
Well, on this line, we should have a `if (!class_exists(Middleware::class)` condition only, so that the `doctrine.dbal.logger` is not added if logging is already handled by the middleware (and then, the `doctrine.dbal.logger.profiling` logger might be injected directly instead of wrapping it in a chain logger if it is alone, as is already the case if you enable profiling without logging)

Reply:
Indeed, my thinking was wrong. Fixed 👍